Maple2.File.Ingest/Mapper/NifMapper.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Maple2.File.IO.Nif;
+using Maple2.File.Parser;
+using Maple2.Model.Metadata;
+
+namespace Maple2.File.Ingest.Mapper;
+
+public class NifMapper : TypeMapper<NifMetadata> {
+    private readonly NifParser nifParser;
+    public Dictionary<uint, NifDocument> nifDocuments;
+    public List<NXSMeshMetadata> nxsMeshes;
```

Comment:
Can you move this (and other related code) to the other Mapper? The data being parsed here is not even used in this file. Iterating the parser twice is not that big of a deal.

Reply:
There's not much that can be moved over because the NifMapper needs to know the NxsMesh data index to store to refer to it. Even if the code for adding to nxsMeshes is moved over most of the code will still be in place. Splitting the 2 like that introduces risks of changes desyncing and breaking the mappings between the tables generated by NifMapper & NxsMeshMapper as well.

It's also a good 50 sec to run the parser over all the nif files on my system.

Reply:
takes ~25s for me but might be a signal that parallel parsing would be worth it. We were able to shave a lot of time down using it for map entities.

My initial thoughts are that we could instead of use a separate class to hold the built index and reuse that index for both mappers.

Reply:
Like the index class would do the parsing & generate the nxs & nif containers? That's an idea I can get behind